packages/superset-ui-plugin-chart-table/src/Table.tsx

```diff
@@ -71,6 +72,44 @@ function getText(value: unknown) {
   return String(value);
 }
 
+type columnWidthMetaDataType = {
+  [key: string]: {
+    maxWidth: number;
+    width: number;
+  };
+};
+
+const columnWidthSelector = createSelector<
```

Comment:
Do you need to define all generics?

Usually I do this

```ts
createSelector(
  (a: AType) => a.b,
  a => a.c,
  (b, c) => ...
);
```
Only define the type of the first argument and it will infer correctly the rest.

Reply:
**Important**

You have to define this selector as class function instead of outside. Otherwise this won't work if you have multiple instance of `TableVis`. They will share the same cache and keep invalidating.

Each table should have `this.columnWidthSelector` instead of single `columnWidthSelector`.